generator/src/main/java/com/scottlogic/deg/generator/GenerateExecute.java

```diff
@@ -30,6 +30,14 @@ public GenerateExecute(GenerationConfig config,
 
     @Override
     public void run() {
+
+        if (config.getDataGenerationType() == GenerationConfig.DataGenerationType.RANDOM
```

Comment:
I'm not sure this is the right place for this logic - I can't exactly say where it should be but it feels odd here?

Reply:
I think its ok for this to be checked here, its the earliest point it can be checked.
This is the 'controller' type for this mode of the generator so it makes sense for it to be making these decisions.